src/internal/connector/sharepoint/list.go

```diff
@@ -91,33 +92,52 @@ func loadSiteLists(
 	listIDs []string,
 ) ([]models.Listable, error) {
 	var (
-		results = make([]models.Listable, 0)
-		errs    error
+		results     = make([]models.Listable, 0)
+		semaphoreCh = make(chan struct{}, fetchChannelSize)
+		errs        error
+		wg          sync.WaitGroup
+		m           sync.Mutex
 	)
+	defer close(semaphoreCh)
+
+	errUpdater := func(id string, err error) {
+		m.Lock()
+		errs = support.WrapAndAppend(id, err, errs)
+		m.Unlock()
+	}
+
+	updateLists := func(list models.Listable) {
+		m.Lock()
+		results = append(results, list)
+		m.Unlock()
+	}
 
 	for _, listID := range listIDs {
-		entry, err := gs.Client().SitesById(siteID).ListsById(listID).Get(ctx, nil)
-		if err != nil {
-			errs = support.WrapAndAppend(
-				listID,
-				errors.Wrap(err, support.ConnectorStackErrorTrace(err)),
-				errs,
-			)
-		}
+		semaphoreCh <- struct{}{}
+		wg.Add(1)
 
-		cols, cTypes, lItems, err := fetchListContents(ctx, gs, siteID, listID)
-		if err == nil {
-			entry.SetColumns(cols)
-			entry.SetContentTypes(cTypes)
-			entry.SetItems(lItems)
-		} else {
-			errs = support.WrapAndAppend("unable to fetchRelationships during loadSiteLists", err, errs)
-			continue
-		}
+		go func(id string) {
+			defer wg.Done()
+			defer func() { <-semaphoreCh }()
 
-		results = append(results, entry)
+			entry, err := gs.Client().SitesById(siteID).ListsById(id).Get(ctx, nil)
+			if err != nil {
+				errUpdater(id, support.ConnectorStackErrorTraceWrap(err, ""))
```

Comment:
we need to return here as the below section attempts to set fields on `entry`
```suggestion
				errUpdater(id, support.ConnectorStackErrorTraceWrap(err, ""))
				return
```

Reply:
Resolved with [36704e3](https://github.com/alcionai/corso/pull/2220/commits/36704e3ed6baeb42bc09520c807e5f9fb26157a3)